src/operun/crm/ldap/ldap_sync.py

```diff
@@ -0,0 +1,289 @@
+
+    def unbind(self):
+        """
+        Unbind connection.
+        """
+        try:
+            self.connection.unbind()
+        except AttributeError:
+            logger.info(_('No connection!'))
+        else:
+            logger.info(_('Disconnected from LDAP server...'))
+
+    # Common Utils
+
+    def list_to_dict(self, list_of_items):
+        """
+        Converts list mappings into dict.
+        """
+        return dict(item.split('|') for item in list_of_items)
+
+    def types_to_sync(self):
+        """
+        Return list of Content-Types to sync.
+        """
+        types = ['Account', 'Contact']
+        return types
+
+    def convert_to_object(self, obj):
+        """
+        Converts object if object has getObject() attribute.
+        """
+        if hasattr(obj, 'getObject'):
+            obj = obj.getObject()
+        elif hasattr(obj, 'object'):
+            obj = obj.object
+        return obj
+
+    def generate_mod_attrs(self, obj):
+        """
+        Generate list of attributes and values for use in LDAP.
+        """
+        mod_attrs = []
+        obj = self.convert_to_object(obj)
+        content_type = obj.Type()
+        if content_type in self.types_to_sync():
+            for field in self.get_fields_to_update(content_type):
+                mod_attrs.append(
+                    (self.get_mapped_field(content_type, field),
+                     str(getattr(obj, field)))
+                )
+        return mod_attrs
+
+    def generate_create_mod_attrs(self, obj):
+        """
+        Generates a mod_attrs list with objectClass and UID.
+        Used in LDAP create object mode.
+        """
+        mod_attrs = self.generate_mod_attrs(obj)
+        content_type = obj.Type()
+        object_uid = obj.UID()
+        class_list = {
+            'Account': 'posixGroup',
+            'Contact': 'inetOrgPerson',
+        }
+        mod_attrs.append(('objectclass', class_list[content_type]))
+        mod_attrs.append(('uid', object_uid))
+        return mod_attrs
+
+    def generate_update_mod_attrs(self, obj):
+        """
+        Generates a mod_attrs list with ldap.MOD_REPLACE attribute.
+        Used in LDAP update object attributes mode.
+        """
+        mod_attrs = self.generate_mod_attrs(obj)
+        return [(ldap.MOD_REPLACE,) + item for item in mod_attrs]
+
+    def generate_ldap_dn(self, obj):
+        """
+        Generate path for LDAP modification.
+        """
+        users_dn = api.portal.get_registry_record(name='operun.crm.users_dn')
+        accounts_dn = api.portal.get_registry_record(
+            name='operun.crm.accounts_dn')
+        obj = self.convert_to_object(obj)
+        content_type = obj.Type()
+        if obj and users_dn or accounts_dn:
+            if content_type == 'Contact':
+                object_title = obj.Title()
+                object_type = obj.type
+                ldap_node = 'cn={0},ou={1}s'.format(object_title, object_type)
```

Comment:
Please create a separate method that returns a mapping of contact_type and ou (ldap). Based on this mapping you can construct the ldap_node.